hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieWriteClient.java

```diff
@@ -1190,7 +1186,7 @@ protected Option<String> inlineScheduleCompaction(Option<Map<String, String>> ex
    * @param logCompactionInstantTime Compaction Instant Time
    * @return Collection of Write Status
    */
-  protected HoodieWriteMetadata<O> logCompact(String logCompactionInstantTime, boolean shouldComplete) {
+  public HoodieWriteMetadata<O> logCompact(String logCompactionInstantTime, boolean shouldComplete) {
```

Comment:
It looks like `#compact` and `#logCompact` are only used by the metadata table.  Now that these are made public they can cause confusion.  Could you file a JIRA ticket revisit whether we should fold the logic into metadata writer only?

Reply:
sure 
https://issues.apache.org/jira/browse/HUDI-9390